osu.Game.Tests/Visual/Online/TestSceneCommentsContainer.cs

```diff
@@ -25,17 +25,27 @@ public class TestSceneCommentsContainer : OsuTestScene
         [Cached]
         private readonly OverlayColourProvider colourProvider = new OverlayColourProvider(OverlayColourScheme.Purple);
 
+        [Cached(typeof(IDialogOverlay))]
+        private readonly DialogOverlay dialogOverlay = new DialogOverlay();
+
         private DummyAPIAccess dummyAPI => (DummyAPIAccess)API;
 
         private CommentsContainer commentsContainer;
 
         [SetUp]
         public void SetUp() => Schedule(() =>
-            Child = new BasicScrollContainer
+        {
+            if (dialogOverlay.Parent != null) Remove(dialogOverlay, false);
```

Comment:
Rather than doing this, the more standard way is to override `Content` and provide a better target to `Add`/`Clear` such that the dependencies remain.

You can see an example here:

https://github.com/ppy/osu/blob/eba070a0f800176c6ea75ea6a83f81d0b19071f1/osu.Game.Tests/Visual/UserInterface/TestSceneModPresetColumn.cs#L32-L62